packages/fast-components-react-msft/src/badge/badge.props.ts

```diff
@@ -0,0 +1,34 @@
+import * as React from "react";
+import {
+    BadgeClassNameContract,
+    ManagedClasses,
+} from "@microsoft/fast-components-class-name-contracts-msft";
+
+export enum BadgeAppearance {
+    accent = "accent",
+    highlight = "highlight",
+    lowlight = "lowlight",
+    lightweight = "lightweight",
+}
```

Comment:
Even though this is an MSFT concept/pattern, I don't think we want to set these from enums. We'll want a more extensible solution here. Another scenario I foresee is wanting the "lightweight" to be a different color than just matching the "foreground" color. Perhaps backplate is a prop that get's passed/set; will need to consider the scenario of the text and a11y as well. But don't think we want to limit the options to just these enums.

Reply:
Confirmed with @Falkicon that we'll want a more extensible solution

Reply:
So, trying to understand the API... backplate would be prop that would take what exactly?

Reply:
I think we need to tease out the API and land something extensible. I'm happy to propose in a solution, but I won't be able to jump on it immediately. The enum approach doesn't represent the type of flexibility that I think we want from this component considering all the variations of backplates, and even variations within various MSFT usages of the text color when no backplate exists.

Reply:
Perhaps we could accept an object that takes `backplateColor` and `textColor`, and also accepts the enum above to provide some "default" settings?

Reply:
Is this over complicating it? If I want to change the CSS I can do that through JSSManager overrides like any other component.

Reply:
I don't think we need a complex object - the component is flat. I think the issue with overrides is that there hasn't been a discussion on what the default states should be here. We simply have a port from a previous internal library.

 I've seen this have 3 or 4 different treatments from one MSFT team alone. The size and weight of the text is consistent as well is the size of backplates, but the text and backplate color regularly varies based on context. I think we need a more flexible solution or at least more thought as to what the defaults should be.

Reply:
@chrisdholt I think not knowing a the defaults is a different issue than how we describe those defaults. At a base level, we have named options of known patterns, and outliers can be created using `jssStyleSheet` props.

If we don't know the defaults then we obviously need to figure those out, but I'm not sure that means named enum values is a bad approach. What are the 3 or 4 that you have seen?

Reply:
synced offline. I'm open to using enums but we need to validate the default states and values first and confirm with @Falkicon.

Reply:
@bheston verifying that these are good default styles
<img width="1127" alt="screen shot 2019-02-07 at 11 05 18 am" src="https://user-images.githubusercontent.com/22480660/52436132-4fabdd80-2ac8-11e9-922f-0a8ee6777b97.png">